superset/viz.py

```diff
@@ -1031,7 +1030,7 @@ def get_data(self, df: pd.DataFrame) -> VizData:
         diff_secs = (end - start).total_seconds()
 
         if domain == "year":
-            range_ = diff_delta.years + 1
+            range_ = end.year - start.year + 1
```

Comment:
Here for example I have 2 dates: 2020-08-01 and 2021-01-01 and the diff_delta.year will be "0" years and + 1 => range_ = 1 and will not display the data from "2021" year

Reply:
good catch!